### PR TITLE
feat(protocol): update deployment scripts based on #15061

### PR DESCRIPTION
**Pull request by davidtaikocha**
Also:
- Made chain ID as the contract address prefix: https://github.com/taikoxyz/taiko-mono/pull/15061#discussion_r1377090039
- Changed the L2 premint ETHs to 1 billion

### CODE REVIEW

**Review by cyberhorsey — APPROVED** [review]

File: packages/protocol/genesis/test_config.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    },
+    {
+      '0x1003ff39d25F2Ab16dBCc18EcE05a9B6154f65F4': 1024
+    }
+  ],
+  get contractAddresses() {
+    return {
+      ProxiedTaikoL2: getConstantAddress(`0${this.chainId}`, 1),
+      ProxiedBridge: getConstantAddress(`0${this.chainId}`, 2),
+      ProxiedERC20Vault: getConstantAddress(`0${this.chainId}`, 3),
```
Comment (dantaik) [inline packages/protocol/genesis/test_config.js]: Lets put the shared components together, and give the small number, 1, 2, 3,...
how about for non-sharable components, we give the 100001, 100002,...

File: packages/protocol/genesis/GenerateGenesis.g.sol [hunk 2]
```diff
@@ -24,36 +24,36 @@ contract TestGenerateGenesis is Test, AddressResolver {
...
     );
     address private owner = configJSON.readAddress(".contractOwner");
     address private admin = configJSON.readAddress(".contractAdmin");
 
-    // uint32 public constant BLOCK_GAS_LIMIT = 30_000_000;
-
     function testContractDeployment() public {
         assertEq(block.chainid, 167);
 
+        checkDeployedCode("ProxiedSingletonERC20Vault");
```
Comment (dantaik) [inline packages/protocol/genesis/GenerateGenesis.g.sol:38]: Lets put all singleton contracts together

File: packages/protocol/genesis/test_config.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    {
+      "0x9eAF5590f2c84912A08de97FA28d0529361Deb9E": 1024,
+    },
+    {
+      "0x1003ff39d25F2Ab16dBCc18EcE05a9B6154f65F4": 1024,
+    },
+  ],
+  get contractAddresses() {
+    return {
+      ProxiedSingletonBridgeSingleton: getConstantAddress(
```
Comment (dantaik) [inline packages/protocol/genesis/test_config.js:84]: Put all singletons together and give them addresses with 1, 2,3,
then for non-singleton contracts, lets start with id 10001, 10002,...

**Review by dantaik — APPROVED** [review]

File: packages/protocol/utils/generate_genesis/taikoL2.ts [hunk 4]
```diff
@@ -24,7 +24,8 @@ export async function deployTaikoL2(
 
     const alloc: any = {};
 
-    let bridgeInitialEtherBalance = ethers.BigNumber.from("2").pow(128).sub(1); // MaxUint128
+    // Premint 1 billion premint ethers in bridge, current Ethereum's supply is ~120.27M.
```
Comment (dantaik) [inline packages/protocol/utils/generate_genesis/taikoL2.ts:27]:
```suggestion
    // Premint 1 billion ethers to the bridge, current Ethereum's supply is ~120.27M.
```

File: packages/protocol/genesis/GenerateGenesis.g.sol (hunk 2 above)
Comment (davidtaikocha) [reply packages/protocol/genesis/GenerateGenesis.g.sol:39]: Moved all singleton contracts checks to `testSingletonContractDeployment` and others to `testNonSingletonContractDeployment`

File: packages/protocol/genesis/GenerateGenesis.g.sol [hunk 5]
```diff
@@ -44,7 +42,8 @@ contract TestGenerateGenesis is Test, AddressResolver {
         checkDeployedCode("ProxiedBridge");
         checkDeployedCode("RegularERC20");
         checkDeployedCode("ProxiedAddressManager");
-        checkDeployedCode("ProxiedSignalService");
+        checkDeployedCode("ProxiedAddressManager");
```
[the remark this answers is not in the document]
Comment (davidtaikocha) [reply packages/protocol/genesis/GenerateGenesis.g.sol:68]: Removed

File: packages/protocol/genesis/test_config.js (hunk 1 above)
Comment (davidtaikocha) [reply packages/protocol/genesis/test_config.js]: Changed

File: packages/protocol/genesis/test_config.js (hunk 3 above)
Comment (davidtaikocha) [reply packages/protocol/genesis/test_config.js]: Changed

File: packages/protocol/genesis/test_config.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    },
+    {
+      "0x1003ff39d25F2Ab16dBCc18EcE05a9B6154f65F4": 1024,
+    },
+  ],
+  get contractAddresses() {
+    // ============ Implementations ============
+    // Singletons
+    return {
+      ProxiedSingletonBridge: getConstantAddress(`0${this.chainId}`, 1),
```
Comment (dantaik) [inline packages/protocol/genesis/test_config.js:86]: @david, can we have an env var and if it is True, then we do not deploy Singleton contracts and assume they are deployed (their addresses are known), if False, we deploy these contracts?

**Pull request merged**